code/__HELPERS/icon_smoothing.dm

```diff
@@ -98,34 +98,60 @@
 			var/adjacencies = calculate_adjacencies(A)
 
 			if(A.smooth & SMOOTH_DIAGONAL)
-				diagonal_smooth(A, adjacencies)
+				A.diagonal_smooth(adjacencies)
 			else
 				cardinal_smooth(A, adjacencies)
 
-/proc/diagonal_smooth(atom/A, adjacencies) //TODO: atom smoothing procs, optimize diagonal per previous state & underlay generation
+/atom/proc/diagonal_smooth(adjacencies)
```

Comment:
Any datum that is supposed to have a physical presence in the ``world`` needs to be a subtype of /atom.

How long have you been coding for byond and you still don't know this.